### PR TITLE
[receiver/hostmetrics] Migrate filesystem scraper to new metrics builder

### PR DESCRIPTION
**Pull request by abhide**
This change migrates hostmetrics/filesystem scraper to the new mdatagen metrics builder.

There are no functional changes on the metrics output except for metrics reordering within the emitted batch.

At this point, this change adds ability to disable collection of particular metrics: to be documented later.

**Link to tracking Issue:** open-telemetry/opentelemetry-collector-contrib#7136

### CODE REVIEW

**Comment by dashpole** [comment]
There isn't much information in the issue to understand what you are trying to do here.  Can you update the issue or the PR description to describe what you are trying to accomplish?

**Comment by dmitryax** [comment]
@dashpole this is a part of a big initiative to migrate scrapers to the new metrics builder to improve configuration options and reduce boilerplate code. The linked issue has more details including design doc https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6429

**Comment by dmitryax** [comment]
@abhide I updated the PR description to make it clear. Let's keep using similar description for other PRs

**Comment by abhide** [comment]
> @abhide I updated the PR description to make it clear. Let's keep using similar description for other PRs

Thanks @dmitryax. Will edit descriptions for other PRs. 

Sorry for the confusion @dashpole

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go [hunk 1]
```diff
@@ -32,10 +34,13 @@ const (
 
 // scraper for FileSystem Metrics
 type scraper struct {
-	config   *Config
-	fsFilter fsFilter
+	config    *Config
+	fsFilter  fsFilter
+	mb        *metadata.MetricsBuilder
+	startTime pdata.Timestamp
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go:40]: Why `startTime` field is needed? Looks like it's not being used

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_test.go [hunk 2]
```diff
@@ -242,10 +279,8 @@ func TestScrape(t *testing.T) {
 func assertFileSystemUsageMetricValid(
 	t *testing.T,
 	metric pdata.Metric,
-	descriptor pdata.Metric,
 	expectedDeviceDataPoints int,
 	expectedDeviceAttributes []map[string]pdata.AttributeValue) {
-	internal.AssertDescriptorEqual(t, descriptor, metric)
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_test.go:248]: Same comment as in another PR, let's validate metric's name.

**Comment by dashpole** [comment]
No worries.  I missed the "Tracked in X" bubble at the top of the issue.

**Pull request closed without merging**